pkg/rbac/userData_test.go

```diff
@@ -578,3 +583,48 @@ func Test_managedCluster_GetUserData(t *testing.T) {
 		t.Errorf("Expected 1 managed cluster but got %d", len(mcResult))
 	}
 }
+
+func Test_getUserData(t *testing.T) {
+
+	mock_cache := mockNamespaceCache()
+	mock_cache = setupToken(mock_cache)
+	managedClusters := make(map[string]struct{})
+	nsRes := make(map[string][]Resource)
+	nsResources := []Resource{{Apigroup: "", Kind: "pods"}}
+	nsRes["ns1"] = nsResources
+	csres := []Resource{{Apigroup: "storage.k8s.io", Kind: "nodes"}}
+	mock_cache = addCSResources(mock_cache, csres)
+
+	//mock cache for cluster-scoped resouces
+
+	allowedres := []Resource{{Apigroup: "storage.k8s.io", Kind: "nodes"}}
+	managedClusters["some-managed-cluster"] = struct{}{}
+	managedClusters["some-other-managed-cluster"] = struct{}{}
+
+	mock_cache.users["unique-user-id"] = &UserDataCache{
+		userData:          UserData{CsResources: allowedres, ManagedClusters: managedClusters, NsResources: nsRes},
+		clustersUpdatedAt: time.Now(),
+		// Using current time , GetUserData should have the same values as cache
+		csrUpdatedAt: time.Now(),
+		nsrUpdatedAt: time.Now(),
+	}
```

Comment:
I find it easier to understand when the mock data is defined inline. Thoughts?

```suggestion
	mock_cache.users["unique-user-id"] = &UserDataCache{
		userData: UserData{
			CsResources:     []Resource{{Apigroup: "storage.k8s.io", Kind: "nodes"}},
			ManagedClusters: map[string]struct{}{"some-managed-cluster": {}, "some-other-managed-cluster": {}},
			NsResources:     map[string][]Resource{"ns1": []Resource{{Apigroup: "", Kind: "pods"}}},
		},
		// Using current time , GetUserData should have the same values as cache
		clustersUpdatedAt: time.Now(),
		csrUpdatedAt:      time.Now(),
		nsrUpdatedAt:      time.Now(),
	}
```

Reply:
The existing tests follow this pattern so I'm also good with consistency.